src/org/zaproxy/zap/extension/tlsdebug/HttpsCallerProcess.java

```diff
@@ -0,0 +1,69 @@
+package org.zaproxy.zap.extension.tlsdebug;
+
+import static org.junit.Assert.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.protocol.Protocol;
+import org.apache.commons.httpclient.protocol.ProtocolSocketFactory;
+import org.junit.Assert;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.network.SSLConnector;
+
+import ch.csnc.extension.httpclient.SSLContextManager;
+
+public class HttpsCallerProcess {
+
+    private static final String HTTPS = "https";
+
+    public static void main(String[] args) {

```

Comment:
The spawned process would have to read the configurations being used by ZAP, it would have to be passed the path to the configuration file as argument to the process (client certificate options are not yet persisted, though).